app-backend/tool/src/main/scala/eval/LazyTile.scala

```diff
@@ -286,4 +295,60 @@ object LazyTile {
     def bind(args: Map[Var, LazyTile]): LazyTile =
       this
   }
+
+  trait FocalOperation extends Tree {
+    val gridbounds: Option[GridBounds]
+    override def cols: Int = gridbounds.map(_.width).getOrElse(left.cols)
+    override def rows: Int = gridbounds.map(_.height).getOrElse(left.rows)
+    def right = LazyTile.Nil
+    def intTile: Tile
+    def dblTile: Tile
+
+    def get(col: Int, row: Int) = intTile.get(col, row)
+    def getDouble(col: Int, row: Int) = dblTile.getDouble(col, row)
+    def bind(args: Map[Var, LazyTile]): LazyTile = this
+  }
+
+  @SuppressWarnings(Array("OptionGet"))
```

Comment:
We could either go with this or fail silently by treating inputs as `NoData`. I'm partial to just letting it throw here and potentially beefing up error reporting with extension of `Either` or `Validated`